### PR TITLE
8 components templates

### PR DESCRIPTION
**Pull request by VladimirSkarb**
https://vladimirskarb.github.io/frontend-course-2020/homeworks/Vladimir.Karbivnychiy_VladimirSkarb/8-components-templates/pictures-slider/dist/pictures-slider/
angular

### CODE REVIEW

**Comment by VladimirSkarb** [comment]
Я сделал бесконечную карусель. После последнего фото, идет первое, и наоборот в обратном порядке.

File: homeworks/Vladimir.Karbivnychiy_VladimirSkarb/8-components-templates/pictures-slider/src/app/components/image-item/image-item.component.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  selector: 'app-image-item',
+  templateUrl: './image-item.component.html',
+  styleUrls: ['./image-item.component.css']
+})
+export class ImageItemComponent {
+
+  @Input() item?: DataInterface;
+  @Output() showEvent: EventEmitter<string> = new EventEmitter();
+
+  loader: any = true;
```
Comment (DrSmile444) [inline homeworks/Vladimir.Karbivnychiy_VladimirSkarb/8-components-templates/pictures-slider/src/app/components/image-item/image-item.component.ts:19]: Don't use `any` in your code

File: homeworks/Vladimir.Karbivnychiy_VladimirSkarb/8-components-templates/pictures-slider/src/app/components/image-gallery/gallery.component.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+  ngOnInit(): void {
+    this.fullImage = this.list[0];
+  }
+
+  showItem(id: string): void {
+    this.fullImage = this.list.find((image: DataInterface) => image.id === id);
+  }
+
+  nextEl(): void {
```
Comment (DrSmile444) [inline homeworks/Vladimir.Karbivnychiy_VladimirSkarb/8-components-templates/pictures-slider/src/app/components/image-gallery/gallery.component.ts:26]: I assume that this method **show** next element. Each function in JS should be named from the verb:
- `showList`
- `hideItem`
- `matchString`
- `replaceAll`

Fix your `nextEl` and `prevEl` names

File: homeworks/Vladimir.Karbivnychiy_VladimirSkarb/8-components-templates/pictures-slider/src/app/components/image-item/image-item.component.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+  constructor() {
+  }
+
+  hiddenLoader(): void {
```
Comment (DrSmile444) [inline homeworks/Vladimir.Karbivnychiy_VladimirSkarb/8-components-templates/pictures-slider/src/app/components/image-item/image-item.component.ts:24]: `hideLoader` sounds better

File: homeworks/Vladimir.Karbivnychiy_VladimirSkarb/8-components-templates/pictures-slider/src/app/components/image-item/image-item.component.ts [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+import {DataInterface} from '../../interfaces/dataInterface';
+
+@Component({
+  selector: 'app-image-item',
+  templateUrl: './image-item.component.html',
+  styleUrls: ['./image-item.component.css']
+})
+export class ImageItemComponent {
+
+  @Input() item?: DataInterface;
```
Comment (DrSmile444) [inline homeworks/Vladimir.Karbivnychiy_VladimirSkarb/8-components-templates/pictures-slider/src/app/components/image-item/image-item.component.ts:16]: `item` is required, isn't it?

File: homeworks/Vladimir.Karbivnychiy_VladimirSkarb/8-components-templates/pictures-slider/src/app/interfaces/dataInterface.ts [hunk 5]
```diff
@@ -0,0 +1,10 @@
+import { UrlsInterface } from './urlsInterface';
+import { LinksInterface } from './linksInterface';
+
+export interface DataInterface {
```
Comment (DrSmile444) [inline homeworks/Vladimir.Karbivnychiy_VladimirSkarb/8-components-templates/pictures-slider/src/app/interfaces/dataInterface.ts:4]: 1) You don't need to add the `Interface` postfix for each interface you create

2) `Data` name is too abstract. `Image` name would be better and easier to understand

3) Move all your image related interfaces into the one file 

**Applicable to all interfaces**

**Review by DrSmile444 — CHANGES_REQUESTED** [review]

File: homeworks/Vladimir.Karbivnychiy_VladimirSkarb/8-components-templates/pictures-slider/src/app/interfaces/Image.ts [hunk 6]
```diff
@@ -0,0 +1,18 @@
+export interface Image {
```
Comment (DrSmile444) [inline homeworks/Vladimir.Karbivnychiy_VladimirSkarb/8-components-templates/pictures-slider/src/app/interfaces/Image.ts:1]: I didn't mean merge all interfaces into the one)
I mean to just move all interfaces into one file. But this is not blocking

**Review by DrSmile444 — APPROVED** [review]

**Pull request merged**